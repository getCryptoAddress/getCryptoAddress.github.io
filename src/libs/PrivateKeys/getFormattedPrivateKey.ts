import { encodePrivateKeyWif, instantiateSha256 } from "@bitauth/libauth";
import type PrivateKeyFormatted from "./types/PrivateKeyFormatted";

export default async function getFormattedPrivateKey(
  randomArray: Uint8Array
): Promise<PrivateKeyFormatted> {
  const sha256 = await instantiateSha256();

  const cache: any = {};

  return Object.freeze({
    get uint8Array() {
      if (!cache["uint8Array"]) {
        cache["uint8Array"] = randomArray;
      }
      return cache["uint8Array"];
    },
    get array() {
      if (!cache["array"]) {
        cache["array"] = Object.freeze(Array.from(randomArray));
      }
      return cache["array"];
    },
    get hex() {
      if (!cache["hex"]) {
        cache["hex"] = Array.from(randomArray, function (byte) {
          return ("0" + (byte & 0xff).toString(16)).slice(-2);
        }).join("");
      }
      return cache["hex"];
    },
    get wifMainnet() {
      if (!cache["wifMainnet"]) {
        cache["wifMainnet"] = encodePrivateKeyWif(
          sha256,
          randomArray,
          "mainnet"
        );
      }
      return cache["wifMainnet"];
    },
    get wifMainnetUncompressed() {
      if (!cache["wifMainnetUncompressed"]) {
        cache["wifMainnetUncompressed"] = encodePrivateKeyWif(
          sha256,
          randomArray,
          "mainnet-uncompressed"
        );
      }
      return cache["wifMainnetUncompressed"];
    },
    get wifTestnet() {
      if (!cache["wifTestnet"]) {
        cache["wifTestnet"] = encodePrivateKeyWif(
          sha256,
          randomArray,
          "testnet"
        );
      }
      return cache["wifTestnet"];
    },
    get wifTestnetUncompressed() {
      if (!cache["wifTestnetUncompressed"]) {
        cache["wifTestnetUncompressed"] = encodePrivateKeyWif(
          sha256,
          randomArray,
          "testnet-uncompressed"
        );
      }
      return cache["wifTestnetUncompressed"];
    },
  });
}
