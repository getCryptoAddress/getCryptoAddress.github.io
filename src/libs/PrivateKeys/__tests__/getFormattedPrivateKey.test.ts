import { describe, expect, it } from "vitest";
import getFormattedPrivateKey from "@/libs/PrivateKeys/getFormattedPrivateKey";

const key1 = new Uint8Array([
  132, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);
const key2 = new Uint8Array([
  133, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);

describe("[PrivateKeys][Util] getFormattedPrivateKey", () => {
  it("Should return correct formats for key1", async () => {
    const formattedKey1 = await getFormattedPrivateKey(key1);
    expect(JSON.stringify(formattedKey1)).toMatchSnapshot();
  });

  it("Should return correct formats for key2", async () => {
    const formattedKey2 = await getFormattedPrivateKey(key2);
    expect(JSON.stringify(formattedKey2)).toMatchSnapshot();
  });
});
