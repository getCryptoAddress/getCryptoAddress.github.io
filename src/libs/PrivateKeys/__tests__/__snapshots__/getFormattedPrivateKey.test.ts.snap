// Vitest Snapshot v1

exports[`[PrivateKeys][Util] getFormattedPrivateKey > validate keys 1`] = `"{\\"uint8Array\\":{\\"0\\":132,\\"1\\":205,\\"2\\":117,\\"3\\":44,\\"4\\":180,\\"5\\":113,\\"6\\":245,\\"7\\":200,\\"8\\":213,\\"9\\":19,\\"10\\":97,\\"11\\":187,\\"12\\":70,\\"13\\":157,\\"14\\":86,\\"15\\":111,\\"16\\":19,\\"17\\":131,\\"18\\":14,\\"19\\":18,\\"20\\":196,\\"21\\":57,\\"22\\":9,\\"23\\":203,\\"24\\":64,\\"25\\":196,\\"26\\":19,\\"27\\":26,\\"28\\":73,\\"29\\":35,\\"30\\":126,\\"31\\":80},\\"array\\":[132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80],\\"hex\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"wifMainnet\\":\\"L1frw5wWqWoj4uKVEgfxYHhTeBqxfyj7yV6dAvu9Yhjo816Es5PS\\",\\"wifMainnetUncompressed\\":\\"5JpmqxVfUkujA3LbVUZqqKjhUTUQVKHnzCob2Lv3qsTv3Vwc6fX\\",\\"wifTestnet\\":\\"cS2rPzwNGaVzELnkd6V5ucCXGR9NLRpp3XF6HMMf3pPoNk79beKj\\",\\"wifTestnetUncompressed\\":\\"92bQRhKD4yys86qt7pTkhvHf87q7eUpzL9fY6yGZBcCxpZPvMn5\\"}"`;

exports[`[PrivateKeys][Util] getFormattedPrivateKey > validate keys 2`] = `"{\\"uint8Array\\":{\\"0\\":133,\\"1\\":205,\\"2\\":117,\\"3\\":44,\\"4\\":180,\\"5\\":113,\\"6\\":245,\\"7\\":200,\\"8\\":213,\\"9\\":19,\\"10\\":97,\\"11\\":187,\\"12\\":70,\\"13\\":157,\\"14\\":86,\\"15\\":111,\\"16\\":19,\\"17\\":131,\\"18\\":14,\\"19\\":18,\\"20\\":196,\\"21\\":57,\\"22\\":9,\\"23\\":203,\\"24\\":64,\\"25\\":196,\\"26\\":19,\\"27\\":26,\\"28\\":73,\\"29\\":35,\\"30\\":126,\\"31\\":80},\\"array\\":[133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80],\\"hex\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"wifMainnet\\":\\"L1hogGAxsNEQJ394wbFaYF24R6xaPdRxrGjhHB3qShPoWVg4XUCk\\",\\"wifMainnetUncompressed\\":\\"5JqDPVMqTN87dwQaX3Ki8XpNAYFrhvZquQ3spAXYpWYSwoT78oT\\",\\"wifTestnet\\":\\"cS4o9BApJRvfTUcLL14huZX83LFz45XevJtAPbWLwp3omEnmXVXU\\",\\"wifTestnetUncompressed\\":\\"92bqyEBP3bCFbzus9PDd18NKpCcZs673FLuptnt4AFHViqoxxEx\\"}"`;
